src/instructlab/sdg/generate_data.py

```diff
@@ -30,7 +29,8 @@
     SynthSkillsFlow,
 )
 from instructlab.sdg.pipeline import Pipeline
-from instructlab.sdg.utils import models
+from instructlab.sdg.sdg import SDG
+from instructlab.sdg.utils import GenerateException, models
```

Comment:
Why not import `GenerateException` from `instructlab.sdg` - i.e. generate_data() uses the API as an external user would use it?